arrows/core/derive_metadata.cxx

```diff
@@ -0,0 +1,350 @@
+  {
+    VITAL_THROW( kv::invalid_value,
+                 "metadata does not contain sensor fov" );
+  }
+
+  auto const x_fov = x_fov_item.as_double() * kv::deg_to_rad;
+  auto const y_fov = y_fov_item.as_double() * kv::deg_to_rad;
+
+  return { x_fov, y_fov };
+}
+
+// ----------------------------------------------------------------------------
+double
+get_slant_range( kwiver::vital::metadata_sptr const& metadata )
+{
+  kv::metadata_item const& item =
+    metadata->find( kv::VITAL_META_SLANT_RANGE );
+
+  if( !item )
+  {
+    VITAL_THROW( kv::invalid_value,
+                 "metadata does not contain slant range" );
+  }
+
+  return item.as_double();
+}
+
+// ----------------------------------------------------------------------------
+kv::geo_point
+get_sensor_location( kwiver::vital::metadata_sptr const& metadata )
+{
+  kv::metadata_item const& item =
+    metadata->find( kv::VITAL_META_SENSOR_LOCATION );
+
+  if( !item )
+  {
+    VITAL_THROW( kv::invalid_value,
+                 "metadata does not contain sensor location" );
+  }
+
+  return kv::any_cast< kv::geo_point >( item.data() );
+}
+
+// ----------------------------------------------------------------------------
+kv::geo_point
+get_frame_center( kwiver::vital::metadata_sptr const& metadata )
+{
+  kv::metadata_item const& item =
+    metadata->find( kv::VITAL_META_FRAME_CENTER );
+
+  if( !item )
+  {
+    VITAL_THROW( kv::invalid_value,
+                 "metadata does not contain frame center" );
+  }
+
+  return kv::any_cast< kv::geo_point >( item.data() );
+}
+
+// ----------------------------------------------------------------------------
+double
+get_target_width( kwiver::vital::metadata_sptr const& metadata )
+{
+  kv::metadata_item const& item =
+    metadata->find( kv::VITAL_META_TARGET_WIDTH );
+
+  if( !item )
+  {
+    VITAL_THROW( kv::invalid_value,
+                 "metadata does not contain target width" );
+  }
+
+  return item.as_double();
+}
+
+// ----------------------------------------------------------------------------
+double
+compute_slant_range( kwiver::vital::metadata_sptr const& metadata )
+{
+  double slant_range = 0.0;
+  try
+  {
+    // Attempt to acquire slant range directly
+    slant_range = get_slant_range( metadata );
+  }
+  catch ( kv::invalid_value const& e )
+  {
+    // Slant range must be calculated from other values
+    kv::rotation_d const total_rotation = get_total_rotation( metadata );
+    double yaw, pitch, roll;
+    total_rotation.get_yaw_pitch_roll( yaw, pitch, roll );
+
+    // Determine the altitude of the sensor above the frame center
+    kv::geo_point const sensor_location = get_sensor_location( metadata );
+    double const sensor_altitude = sensor_location.location()[ 2 ];
+    kv::geo_point const frame_center = get_frame_center( metadata );
+    double const frame_center_altitude = frame_center.location()[ 2 ];
+    double const altitude_difference = sensor_altitude - frame_center_altitude;
```

Comment:
I was initially concerned this was wrong because the `SENSOR_LOCATION` claims to have an altitude while the `FRAME_CENTER` claims to have an elevation. I read the former as height above the ground while the later was height above sea level. However, a read of the [MISB standard ](https://gwg.nga.mil/misb/docs/standards/ST0601.6.pdf) suggests the code is correct and these two terms are actually synonymous. Perhaps adding some comment here would be useful, or additionally updating the documentation in `metadata_tags.h` would be useful.